runtime/compiler/optimizer/JProfilingRecompLoopTest.cpp

```diff
@@ -0,0 +1,267 @@
+ *             |ifcmple goto remainingCodeBlock              | <---- CalculateCheckLoopFreqBlock   |
+ *             |   rawFrequencyOfTheLoop                     |                                     |
+ *             |   AdjustedLoopRecompilationThreshold        |                                     |
+ *             -----------------------------------------------                                     |
+ *                                  |                                                              |
+ *                                  |                                                              |
+ *                                  V                                                              |
+ *             -----------------------------------------------                                     |
+ *             |call TR_jitRetranslateCallerWithPrep         | <--- callRecompilation block        |
+ *             |       loadaddr startPC                      |                                     |
+ *             |       loadaddr                              |                                     |
+ *             -----------------------------------------------                                     |
+ *                                  |                                                              |
+ *                                  |                                                              |
+ *                                  V                                                              |
+ *                   --------------------------------                                              |
+ *                   |treeTops after insertion Point|  <--- remainingCodeBlock                     |
+ *                   |from original block           |                                              |
+ *                   |...                           |<----------------------------------------------   
+ *                   --------------------------------
+ *    \param comp Current compilation object
+ *    \param testLocations RecompilationTestLocation list containing TreeTops after which test are required to be added
+ *                         And corresponding to that location, a loop nesting depth 
+ */
+
+void 
+TR_JProfilingRecompLoopTest::addRecompilationTests(TR::Compilation *comp, RecompilationTestLocations &testLocations)
+   {
+   TR_PersistentProfileInfo *profileInfo = comp->getRecompilationInfo()->findOrCreateProfileInfo();
+   TR_BlockFrequencyInfo *bfi = TR_BlockFrequencyInfo::get(profileInfo);
+   TR::CFG *cfg = comp->getFlowGraph();
+   cfg->setStructure(NULL);
+
+   // Following environment sets up the base recompilation threshold for the loop.
+   // This base recompile threshold in conjunction with the depth in loop is compared with the raw count of the
+   // loop to decide if we have run this loop enough time to trip method recompilation. 
+   // TODO: Currently set base recompilation threshold needs some tuning. 
```

Comment:
Not sure what this means - is it really a TODO or should this be some advice that you should tune XX parameter if you see such and such a condition?

Reply:
Removed TODO: I believe a method invocation 4000 and loop base threshold 2000 is tuned number that gives us the better performance